cmd/precise-code-intel/src/shared/config/config.ts

```
@@ -67,15 +67,21 @@ function requireRestart(oldConfiguration: Configuration, newConfiguration: Confi
  */
 async function updateConfiguration(logger: Logger, onChange: (configuration: Configuration) => void): Promise<never> {
     const start = Date.now()
+    let previousError: Error | undefined
     while (true) {
         try {
             onChange(await loadConfiguration())
+            previousError = undefined
         } catch (error) {
-            // Suppress log messages for errors caused by the frontend being unreachable until we've
-            // given the frontend enough time to initialize (in case other services start up before
-            // the frontend), to reduce log spam.
-            if (Date.now() - start > settings.DELAY_BEFORE_UNREACHABLE_LOG * 1000 || error.code !== 'ECONNREFUSED') {
+            if (
+                error.code !== 'ECONNREFUSED' &&
```

I think this would ignore all errors that are not `ECONNREFUSED`, which is probably not what was intended?

My idea would be to leave this clause out of this `if` here (so all errors are throttled), but inside, if it's only `ECONNREFUSED`, only log `error.message` instead of the whole thing (which might still be useful for unexpected errors). That would address @slimsag's ask to have it only be a single line for this common case